nemo/collections/llm/tools/auto_configurator/core/training_config.py

```diff
@@ -0,0 +1,1097 @@
+        elif model_size_in_b <= 4.0:
+            tp = [1, 2, 4]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            mbs = [1, 2, 4]
+            gbs = 128
+        elif model_size_in_b <= 8.0:
+            tp = [2, 4]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            mbs = [1, 2]
+            gbs = 256
+        elif model_size_in_b <= 13.0:
+            tp = [2, 4]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            mbs = [1, 2]
+            gbs = 256
+        elif model_size_in_b <= 23.0:
+            tp = [4, 8]
+            pp = [x for x in valid_pp if 1 <= x <= 4]
+            mbs = [1]
+            min_model_parallel = 8
+            max_model_parallel = 32
+            gbs = 256
+        elif model_size_in_b <= 45.0:
+            tp = [8]
+            pp = [x for x in valid_pp if 4 <= x <= 8]
+            mbs = [1]
+            min_model_parallel = 32
+            max_model_parallel = 64
+            gbs = 256
+    elif seq_length == 16384:
+        if model_size_in_b <= 1.0:
+            tp = [2, 4]
+            mbs = [1, 2]
+            gbs = 32
+        elif model_size_in_b <= 4.0:
+            tp = [2, 4]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            mbs = [1]
+            gbs = 64
+        elif model_size_in_b <= 8.0:
+            tp = [2, 4]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            mbs = [1]
+            gbs = 128
+        elif model_size_in_b <= 13.0:
+            tp = [2, 4]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            mbs = [1]
+            gbs = 128
+        elif model_size_in_b <= 23.0:
+            tp = [4, 8]
+            pp = [x for x in valid_pp if 2 <= x <= 4]
+            mbs = [1]
+            min_model_parallel = 8
+            max_model_parallel = 32
+            gbs = 128
+    elif seq_length == 32768:
+        if model_size_in_b <= 1.0:
+            tp = [2, 4]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            mbs = [1]
+            gbs = 16
+        elif model_size_in_b <= 4.0:
+            tp = [2, 4]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            mbs = [1]
+            gbs = 32
+        elif model_size_in_b <= 8.0:
+            tp = [4, 8]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            min_model_parallel = 4
+            max_model_parallel = 16
+            mbs = [1]
+            gbs = 64
+        elif model_size_in_b <= 13.0:
+            tp = [4, 8]
+            pp = [x for x in valid_pp if 1 <= x <= 2]
+            min_model_parallel = 4
+            max_model_parallel = 16
+            mbs = [1]
+            gbs = 64
+        elif model_size_in_b <= 23.0:
+            tp = [8]
+            pp = [x for x in valid_pp if 2 <= x <= 4]
+            mbs = [1]
+            min_model_parallel = 16
+            max_model_parallel = 32
+            gbs = 64
+
+    return tp, pp, cp, ep, mbs, min_model_parallel, max_model_parallel, gbs
```

Comment:
This doesn't match with the type-hint. Returning a tuple of length 8 makes the code very hard to follow, please turn this into a dataclass and return that. You can do that same new dataclass in the other functions in this file.